test/easyconfigs/easyconfigs.py

```diff
@@ -1333,21 +1333,29 @@ def template_easyconfig_test(self, spec):
 
     # make sure all patch files are available
     specdir = os.path.dirname(spec)
+    basedir = os.path.dirname(os.path.dirname(specdir))
     specfn = os.path.basename(spec)
     for idx, patch in enumerate(ec['patches']):
         if isinstance(patch, (tuple, list)):
-            patch = patch[0]
+            patch_name = patch[0]
+            patch_dir = specdir
+        elif isinstance(patch, dict):
+            patch_name = patch['name']
+            if patch['alt_location']:
+                patch_dir = os.path.join(basedir, letter_dir_for(patch_name), patch_name)
+            else:
+                patch_dir = specdir
 
         # only check actual patch files, not other files being copied via the patch functionality
-        patch_full = os.path.join(specdir, patch)
-        if patch.endswith('.patch'):
+        patch_full = os.path.join(patch_dir, patch_name)
```

Comment:
Are patch_dir and patch_name correctly defined here if the patch instance is a string?

Reply:
Ahh, probably not... will fix that